tensorflow/compiler/xla/mlir_hlo/tosa/transforms/legalize_mhlo/legalize_mhlo.cc

```diff
@@ -17,14 +17,15 @@ limitations under the License.
 #include <utility>
 
 #include "mhlo/IR/hlo_ops.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
 #include "mlir/Dialect/Tosa/IR/TosaOps.h"
 #include "mlir/IR/BuiltinAttributes.h"
 #include "mlir/Parser/Parser.h"
 #include "mlir/Transforms/GreedyPatternRewriteDriver.h"
 #include "transforms/passes.h"
 
 #define GEN_PASS_DEF_TOSALEGALIZEMHLOPASS
-#include "transforms/passes.h.inc"
+#include "tosa/transforms/passes.h.inc"
```

Comment:
The error is: fatal error: 'tosa/transforms/passes.h.inc' file not found
The bazel build expects the header include exactly as transforms/passes.h.inc.
Is it possible to change CMake build instead, so that it will work with the transforms/passes.h.inc include?
In any case, I am not familiar with neither how the bazel BUILD is setup in this case nor with the CMake setup, maybe @jpienaar can help here?

Reply:
The CMake build breaks if I revert to including just `transforms/passes.h.inc`.  I imagine that the problem is with the Bazel build since `BUILD` doesn't seem to contain a target to run TableGen on `tosa/transforms/passes.td`.

Sadly, I don't know Bazel well enough to reproduce and fix the build, so perhaps the best way out is to drop the TOSA pipeline from the CMake build rules as well.  If you agree, I'll close this PR and send a new one for changing the CMake rules.

Reply:
> The CMake build breaks if I revert to including just `transforms/passes.h.inc`. I imagine that the problem is with the Bazel build since `BUILD` doesn't seem to contain a target to run TableGen on `tosa/transforms/passes.td`.
> 
> Sadly, I don't know Bazel well enough to reproduce and fix the build, so perhaps the best way out is to drop the TOSA pipeline from the CMake build rules as well. If you agree, I'll close this PR and send a new one for changing the CMake rules.

The Bazel build is fine, here is where the transforms/passes.h.inc file is generated:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/mlir_hlo/tosa/BUILD#L46

So transforms/passes.h.inc file exists in the Bazel build and is successfully found. If you change the path, it is not found anymore, and the build breaks.

Reply:
I'm guessing it is missing include directory added and seems like it's bazel side (at least transforms/passes.h.inc would be ambiguous without tosa prefix here). Unfortunately person who is working on this is out for a bit. I'll try and look, but I think it probably boils down to missing include dir.

Reply:
(it'll be a couple of hours before I can at best due to other responsibilities)

Reply:
I think I have a way to keep both builds happy without too many changes (we need a bit of refactoring to fit closer to upstream style if we want to go beyond that). Testing it out now with CMake. Don't know how you prefer to proceed, I could create a new PR or

Reply:
Thanks Jacques, much appreciated!  Feel free to create a separate PR and close this one.